contrib/world_builder/include/world_builder/world.h

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright (C) 2018 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef _world_builder_world_h
+#define _world_builder_world_h
+
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <world_builder/features/interface.h>
+#include <world_builder/coordinate_systems/interface.h>
+
+using boost::property_tree::ptree;
+
+
+namespace WorldBuilder
+{
+  namespace Features
+  {
+    class Interface;
+  }
+
+  class World
+  {
+    public:
+      /**
+       * constructor
+       */
+      World(std::string filename);
+
+      /**
+       * Destructor
+       */
+      ~World();
+
+      /**
+       * read in the world builder file
+       */
+      void read(ptree &property_tree);
+
+      double temperature(const std::array<double, 2> point, const double depth, const double gravity_norm) const;
+
+      double temperature(const std::array<double, 3> point, const double depth, const double gravity_norm) const;
```

Comment:
what is the difference of the 2 and 3 version?

Reply:
The 2d version converts the 2d point to a 3d point and calls the 3d function.